src/stores/AccountController.ts

```diff
@@ -359,57 +369,72 @@ export const changePasswordWithToken = async (customerEmail: string, newPassword
+          customerId,
+          offerId: subscription.offerId,
+          status,
+          unsubscribeUrl: subscription.unsubscribeUrl,
+        },
+        sandbox,
+        jwt,
+      );
 
-    if (response.errors.length > 0) throw new Error(response.errors[0]);
+      if (response.errors.length > 0) throw new Error(response.errors[0]);
 
-    await reloadActiveSubscription();
+      await reloadActiveSubscription();
 
-    return response.responseData;
+      return response.responseData;
+    });
   });
 };
 
-export async function reloadActiveSubscription({ delay }: { delay: number } = { delay: 0 }) {
-  useAccountStore.setState({ loading: true });
+export async function checkEntitlements(offerId?: string): Promise<unknown> {
+  return await useAccount(async ({ auth: { jwt } }) => {
+    return await useService(async ({ checkoutService, sandbox }) => {
+      if (!offerId) {
+        return false;
+      }
+      const response = await checkoutService.getEntitlements({ offerId }, sandbox, jwt);
+      return !!response;
+    });
+  });
+}
 
-  return await useLoginContext(async ({ cleengSandbox, customerId, auth: { jwt } }) => {
-    const [activeSubscription, transactions, activePayment] = await Promise.all([
-      getActiveSubscription({ cleengSandbox, customerId, jwt }),
-      getAllTransactions({ cleengSandbox, customerId, jwt }),
-      getActivePayment({ cleengSandbox, customerId, jwt }),
-    ]);
+export async function reloadActiveSubscription({ delay }: { delay: number } = { delay: 0 }): Promise<unknown> {
+  useAccountStore.setState({ loading: true });
+  return await useAccount(async ({ customerId, auth: { jwt } }) => {
+    return await useService(async ({ subscriptionService, sandbox, config }) => {
+      // The subscription data takes a few seconds to load after it's purchased,
+      // so here's a delay mechanism to give it time to process
+      if (delay > 0) {
+        return new Promise((resolve: (value?: unknown) => void) => {
+          window.setTimeout(() => {
+            reloadActiveSubscription().finally(resolve);
+          }, delay);
+        });
+      }
 
-    // The subscription data takes a few seconds to load after it's purchased,
-    // so here's a delay mechanism to give it time to process
-    if (delay > 0) {
-      return new Promise((resolve: (value?: unknown) => void) => {
-        window.setTimeout(() => {
-          reloadActiveSubscription().finally(resolve);
-        }, delay);
+      const [activeSubscription, transactions, activePayment] = await Promise.all([
+        subscriptionService.getActiveSubscription({ sandbox, customerId, jwt, config }),
+        subscriptionService.getAllTransactions({ sandbox, customerId, jwt }),
+        subscriptionService.getActivePayment({ sandbox, customerId, jwt }),
+      ]);
+      // this invalidates all entitlements caches which makes the useEntitlement hook to verify the entitlements.
+      await queryClient.invalidateQueries('entitlements');
```

Comment:
This will be slightly easier to follow if `'entitlements'` is a string constant shared here and where the queries are

Reply:
This is an existing code, only the order in the function is updated.